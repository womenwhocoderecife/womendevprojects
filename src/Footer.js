import React from 'react';

import logo from './assets/images/logo.png';
import github from './assets/icons/github.svg';
import instagram from './assets/icons/instagram.svg';
import twitter from './assets/icons/twitter.svg';
import logoWwcode from './assets/images/logo-wwcode.png';

import './assets/style/footer.css';

const Footer = () => (
  <footer className="footer">
    <img className="footer__img" src={logo} alt="logo" />

    <ul className="footer__list">
      <li className="footer__list__item">
        <h3 className="footer__list__title">Sobre</h3>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          O que é
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Como Participar
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          FAQ
        </a>
      </li>
    </ul>

    <ul className="footer__list">
      <li className="footer__list__item">
        <h3 className="footer__list__title">Projetos</h3>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Pro-Bonos
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Comerciais
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Clientes
        </a>
      </li>
    </ul>

    <div className="footer__list">
      <h3 className="footer__list__title footer__title">
        Mídia Social
      </h3>
      <ul className="footer__list__social-network">
        <li className="footer__list__item__social-network">
          <a href="https://github.com/womenwhocoderecife">
            <img src={github} alt="logo" />
          </a>
        </li>
        <li className="footer__list__item__social-network">
          <a href="https://www.instagram.com/wwcoderecife/">
            <img src={instagram} alt="logo" />
          </a>
        </li>
        <li className="footer__list__item__social-network">
          <a href="https://twitter.com/WWCode_Recife">
            <img src={twitter} alt="logo" />
          </a>
        </li>
      </ul>
    </div>

    <ul className="footer__list">
      <li className="footer__list__wwcode">
        <h3 className="footer__list__title">Iniciativa</h3>
      </li>
      <li>
        <a href="https://www.womenwhocode.com/recife">
          <img
            className="footer__img"
            src={logoWwcode}
            alt="logo WWCode"
          />
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
