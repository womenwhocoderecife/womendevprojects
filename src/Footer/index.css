.footer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 50px;
  max-width: 1200px;
  width: 100%;
  margin: 0 auto;

  grid-column: 1 / -1;
}

@media (min-width: 760px) {
  .footer {
    flex-direction: row;
    align-items: flex-start;
    justify-content: space-around;
  }
}

.footer__figure {
  margin: 0;
}

.footer__img {
  width: 150px;
  margin-top: -10px;
}

.footer__list {
  list-style: none;
  padding: 0;
  margin: 0 0 30px 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

@media (min-width: 760px) {
  .footer__list {
    align-items: baseline;
    justify-content: flex-end;
    margin-bottom: 0;
  }
}

.footer__list__title {
  margin: 0;
  /* color: var(--dark-primary-color); */
  color: #47525e;
  font-weight: 300 !important;
  font: 500 1em 'Roboto', sans-serif;
  text-decoration: none;
}

.footer__list__item {
  margin-top: 5px;
  padding: 5px;
}

.footer__list__link {
  text-decoration: none;
  color: var(--secondary-color);
  font: 300 14px 'Roboto', sans-serif;
}

.footer__title {
  margin-top: 10px;
  color: var(--dark-primary-color);
  font: 500 1em 'Roboto', sans-serif;
}

.footer__list__social-network {
  display: flex;
  list-style: none;
  padding: 0;
  margin: 0;
}

.footer__list__item__social-network {
  margin: 5px 5px 0 0;
}

.footer__list__wwcode {
  margin: 0px 0 0 10px;
  color: var(--dark-primary-color);
  font: 500 1em 'Roboto', sans-serif;
}
