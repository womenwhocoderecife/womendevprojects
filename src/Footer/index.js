import React from 'react';

import logo from '../assets/images/logo.png';
import github from '../assets/icons/github.svg';
import instagram from '../assets/icons/instagram.svg';
import twitter from '../assets/icons/twitter.svg';
import logoWwcode from '../assets/images/logo-wwcode.png';

import './index.css';

const Footer = () => (
  <footer className="footer">
    <figure className="footer__figure">
      <img
        className="footer__img"
        src={logo}
        alt="logo"
        title="Women Dev Project"
      />
    </figure>

    <ul className="footer__list">
      <li className="footer__list__item">
        <a className="footer__list__title" href="#home">Home</a>
      </li>
      {/* <li className="footer__list__item">
        <a className="footer__list__link" href="">
          O que é
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Como Participar
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          FAQ
        </a>
      </li> */}
    </ul>

    <ul className="footer__list">
      <li className="footer__list__item">
        <a className="footer__list__title" href="#about">Sobre</a>
      </li>
      {/* <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Pro-Bonos
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Comerciais
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Clientes
        </a>
      </li> */}
    </ul>


    <ul className="footer__list">
      <li className="footer__list__item">
        <a className="footer__list__title" href="#proposal">Nossa Proposta</a>
      </li>
      {/* <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Pro-Bonos
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Comerciais
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Clientes
        </a>
      </li> */}
    </ul>

    <ul className="footer__list">
      <li className="footer__list__item">
        <a className="footer__list__title" href="#team">Equipe</a>
      </li>
      {/* <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Pro-Bonos
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Comerciais
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Clientes
        </a>
      </li> */}
    </ul>

    <ul className="footer__list">
      <li className="footer__list__item">
        <a className="footer__list__title" href="#projects">Projetos</a>
      </li>
      {/* <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Pro-Bonos
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Comerciais
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Clientes
        </a>
      </li> */}
    </ul>


    <ul className="footer__list">
      <li className="footer__list__item">
        <a className="footer__list__title" href="#contact">Contato</a>
      </li>
      {/* <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Pro-Bonos
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Comerciais
        </a>
      </li>
      <li className="footer__list__item">
        <a className="footer__list__link" href="">
          Clientes
        </a>
      </li> */}
    </ul>

    <div className="footer__list">
      <h3 className="footer__title">
        Mídia Social
      </h3>
      <ul className="footer__list__social-network">
        <li className="footer__list__item__social-network">
          <a href="https://github.com/womenwhocoderecife">
            <figure className="footer__figure">
              <img src={github} alt="logo" title="github" />
            </figure>
          </a>
        </li>
        <li className="footer__list__item__social-network">
          <a href="https://www.instagram.com/wwcoderecife/">
            <figure className="footer__figure">
              <img src={instagram} alt="logo" title="instagram" />
            </figure>
          </a>
        </li>
        <li className="footer__list__item__social-network">
          <a href="https://twitter.com/WWCode_Recife">
            <figure className="footer__figure">
              <img src={twitter} alt="logo" title="twitter" />
            </figure>
          </a>
        </li>
      </ul>
    </div>

    <ul className="footer__list">
      <li className="footer__list__wwcode">
        <h3 className="footer__title">Iniciativa</h3>
      </li>
      <li>
        <a href="https://www.womenwhocode.com/recife">
          <figure className="footer__figure">
            <img
              className="footer__img"
              src={logoWwcode}
              alt="logo WWCode"
              title="logo WWCode"
            />
          </figure>
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
