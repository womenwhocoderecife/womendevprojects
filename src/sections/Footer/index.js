import React from 'react';

import logo from '../../assets/images/logo.png';
import github from '../../assets/icons/github.svg';
import instagram from '../../assets/icons/instagram.svg';
import twitter from '../../assets/icons/twitter.svg';
import logoWwcode from '../../assets/images/logo-wwcode.png';

import {
  Container,
  WWCodeLogo,
  WWCodeImage,
  ListMenu,
  ListItem,
  ListItemLink,
  SocialMedia,
  SocialMediaTitle,
  SocialMediaList,
  SocialMediaListItem,
  Iniciative,
  IniciativeTitle,
} from './styles';

export default function Footer() {
  return (
    <Container>
      <WWCodeLogo>
        <WWCodeImage
          src={logo}
          alt="logo"
          title="Women Dev Project"
        />
      </WWCodeLogo>

      <ListMenu>
        <ListItem>
          <ListItemLink href="#home">Home</ListItemLink>
        </ListItem>
        <ListItem>
          <ListItemLink href="#about">Sobre</ListItemLink>
        </ListItem>
        <ListItem>
          <ListItemLink href="#proposal">Nossa Proposta</ListItemLink>
        </ListItem>
        <ListItem>
          <ListItemLink href="#projects">Projetos</ListItemLink>
        </ListItem>
        <ListItem>
          <ListItemLink href="#team">Equipe</ListItemLink>
        </ListItem>
        <ListItem>
          <ListItemLink href="#contact">Contato</ListItemLink>
        </ListItem>
      </ListMenu>

      <SocialMedia>
        <SocialMediaTitle>MÃ­dia Social</SocialMediaTitle>
        <SocialMediaList>
          <SocialMediaListItem>
            <a
              href="https://github.com/womenwhocoderecife"
              target="_blank"
              rel="noopener noreferrer"
            >
              <figure>
                <img src={github} alt="logo" title="github" />
              </figure>
            </a>
          </SocialMediaListItem>

          <SocialMediaListItem>
            <a
              href="https://www.instagram.com/wwcoderecife/"
              target="_blank"
              rel="noopener noreferrer"
            >
              <figure>
                <img src={instagram} alt="logo" title="instagram" />
              </figure>
            </a>
          </SocialMediaListItem>

          <SocialMediaListItem>
            <a
              href="https://twitter.com/WWCode_Recife"
              target="_blank"
              rel="noopener noreferrer"
            >
              <figure>
                <img src={twitter} alt="logo" title="twitter" />
              </figure>
            </a>
          </SocialMediaListItem>
        </SocialMediaList>
      </SocialMedia>

      <Iniciative>
        <IniciativeTitle>Iniciativa</IniciativeTitle>

        <a
          href="https://www.womenwhocode.com/recife"
          target="_blank"
          rel="noopener noreferrer"
        >
          <WWCodeLogo>
            <WWCodeImage
              src={logoWwcode}
              alt="logo WWCode"
              title="logo WWCode"
            />
          </WWCodeLogo>
        </a>
      </Iniciative>
    </Container>
  );
}
