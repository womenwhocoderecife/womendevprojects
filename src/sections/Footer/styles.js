import styled from 'styled-components';

export const Container = styled.footer`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 50px;
  max-width: 1200px;
  width: 100%;
  margin: 0 auto;

  grid-column: 1 / -1;

  flex-direction: row;
  align-items: flex-start;
  justify-content: space-around;

  @media (max-width: 800px) {
    display: flex;
    -ms-flex-direction: column;
    flex-direction: column;
    -ms-flex-pack: center;
    justify-content: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 50px;
    max-width: 1200px;
    width: 100%;
    margin: 0 auto;
    grid-column: 1/-1;
  }
`;

export const WWCodeLogo = styled.figure`
  margin: 0;
`;

export const WWCodeImage = styled.img`
  width: 150px;
  margin-top: -10px;
`;

export const ListMenu = styled.ul`
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  padding-left: 0;
  list-style: none;
  margin: 0;

  @media (max-width: 800px) {
    display: flex;
    flex-direction: column;
    align-items: center;
  }
`;

export const ListItem = styled.li`
  margin-top: 5px;
  padding: 5px;
`;

export const ListItemLink = styled.a`
  text-decoration: none;
  color: ${(props) => props.theme.colors.secondary};
  font: 300 1em 'Roboto', sans-serif;

  :hover {
    border-bottom: 1px solid
      ${(props) => props.theme.colors.darkPrimary};
  }
`;

export const SocialMedia = styled.div`
  align-items: baseline;
  justify-content: flex-end;
  margin-bottom: 0;
`;

export const SocialMediaTitle = styled.h3`
  color: ${(props) => props.theme.colors.darkPrimary};
  font: 500 1em Roboto, sans-serif;
  margin: 0;
  padding-top: 10px;
`;

export const SocialMediaList = styled.ul`
  display: flex;
  list-style: none;
  padding: 0;
  margin: 0;
`;

export const SocialMediaListItem = styled.li`
  margin: 5px 5px 0 0;

  figure {
    margin: 0;
  }

  figure:hover {
    border-bottom: 1px solid
      ${(props) => props.theme.colors.darkPrimary};
  }
`;

export const Iniciative = styled.div`
  display: grid;
`;

export const IniciativeTitle = styled(SocialMediaTitle)`
  padding-left: 10px;
  margin-bottom: 5px;
  @media (max-width: 800px) {
    text-align: center;
  }
`;
