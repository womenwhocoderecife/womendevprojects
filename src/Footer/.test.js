import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Footer from '.';

describe('Footer', () => {
  it('renders without crashing', () => {
    const footer = document.createElement('footer');
    ReactDOM.render(<Footer />, footer);
    ReactDOM.unmountComponentAtNode(footer);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
