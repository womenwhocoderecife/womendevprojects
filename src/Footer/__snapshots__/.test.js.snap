// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer has a valid snapshot 1`] = `
<footer
  className="footer"
>
  <figure
    className="footer__figure"
  >
    <img
      alt="logo"
      className="footer__img"
      src="logo.png"
      title="Women Dev Project"
    />
  </figure>
  <ul
    className="footer__list"
  >
    <li
      className="footer__list__item"
    >
      <h3
        className="footer__list__title"
      >
        Sobre
      </h3>
    </li>
    <li
      className="footer__list__item"
    >
      <a
        className="footer__list__link"
        href=""
      >
        O que é
      </a>
    </li>
    <li
      className="footer__list__item"
    >
      <a
        className="footer__list__link"
        href=""
      >
        Como Participar
      </a>
    </li>
    <li
      className="footer__list__item"
    >
      <a
        className="footer__list__link"
        href=""
      >
        FAQ
      </a>
    </li>
  </ul>
  <ul
    className="footer__list"
  >
    <li
      className="footer__list__item"
    >
      <h3
        className="footer__list__title"
      >
        Projetos
      </h3>
    </li>
    <li
      className="footer__list__item"
    >
      <a
        className="footer__list__link"
        href=""
      >
        Pro-Bonos
      </a>
    </li>
    <li
      className="footer__list__item"
    >
      <a
        className="footer__list__link"
        href=""
      >
        Comerciais
      </a>
    </li>
    <li
      className="footer__list__item"
    >
      <a
        className="footer__list__link"
        href=""
      >
        Clientes
      </a>
    </li>
  </ul>
  <div
    className="footer__list"
  >
    <h3
      className="footer__list__title footer__title"
    >
      Mídia Social
    </h3>
    <ul
      className="footer__list__social-network"
    >
      <li
        className="footer__list__item__social-network"
      >
        <a
          href="https://github.com/womenwhocoderecife"
        >
          <figure
            className="footer__figure"
          >
            <img
              alt="logo"
              src="github.svg"
              title="github"
            />
          </figure>
        </a>
      </li>
      <li
        className="footer__list__item__social-network"
      >
        <a
          href="https://www.instagram.com/wwcoderecife/"
        >
          <figure
            className="footer__figure"
          >
            <img
              alt="logo"
              src="instagram.svg"
              title="instagram"
            />
          </figure>
        </a>
      </li>
      <li
        className="footer__list__item__social-network"
      >
        <a
          href="https://twitter.com/WWCode_Recife"
        >
          <figure
            className="footer__figure"
          >
            <img
              alt="logo"
              src="twitter.svg"
              title="twitter"
            />
          </figure>
        </a>
      </li>
    </ul>
  </div>
  <ul
    className="footer__list"
  >
    <li
      className="footer__list__wwcode"
    >
      <h3
        className="footer__list__title"
      >
        Iniciativa
      </h3>
    </li>
    <li>
      <a
        href="https://www.womenwhocode.com/recife"
      >
        <figure
          className="footer__figure"
        >
          <img
            alt="logo WWCode"
            className="footer__img"
            src="logo-wwcode.png"
            title="logo WWCode"
          />
        </figure>
      </a>
    </li>
  </ul>
</footer>
`;
